libvast/src/detail/series_builders.cpp

```diff
@@ -330,7 +330,7 @@ auto series_builder::remove_last_row() -> void {
     [this](auto& actual) {
       if constexpr (std::is_same_v<bool, decltype(actual.remove_last_row())>) {
         if (auto all_values_are_null = actual.remove_last_row();
-            all_values_are_null) {
+            all_values_are_null and can_change_builder_type_) {
```

Comment:
If we can't change the builder type, then it means we have a fixex type for a builder. There is no need to switch it back into an unknown type and allow it to infer

Reply:
I will be adding additional unit tests for this branch in the next commit

Reply:
This makes sense.